packages/components/src/utils/useContainerQueries.spec.tsx

```diff
@@ -0,0 +1,209 @@
+import React from 'react'
+import { act, render, screen } from '@testing-library/react'
+import { vi } from 'vitest'
+import { useContainerQueries } from './useContainerQueries'
+
+const ExampleComponent = (): JSX.Element => {
+  const { containerRef, queries } = useContainerQueries()
+
+  return (
+    <div ref={containerRef} data-testid="container">
```

Comment:
Consumer would need to set `containerRef` on the container that they want to respond to.

This is similar to setting `className="@container"`